pkg/operator/ceph/object/admin.go

```diff
@@ -149,3 +168,46 @@ func isInvalidFlagError(err error) bool {
 	// it's also returned from some other failures, but this should be rare for Rook
 	return exitCode == 22
 }
+
+func GetAdminOPSUserCredentials(ctx *clusterd.Context, clusterInfo *cephclient.ClusterInfo, objContext *Context, cephObjectStore *cephv1.CephObjectStore) (string, string, error) {
+	if cephObjectStore.Spec.IsExternal() {
+		// Fetch the secret for admin ops user
+		s := &v1.Secret{}
+		err := ctx.Client.Get(context.TODO(), types.NamespacedName{Name: RGWAdminOpsUserSecretName, Namespace: cephObjectStore.Namespace}, s)
+		if err != nil {
+			return "", "", err
+		}
+
+		accessKey, ok := s.Data[rgwAdminOpsUserAccessKey]
+		if !ok {
+			return "", "", errors.Errorf("failed to find accessKey %q for rgw admin ops in secret %q", rgwAdminOpsUserAccessKey, RGWAdminOpsUserSecretName)
+		}
+		secretKey, ok := s.Data[rgwAdminOpsUserSecretKey]
+		if !ok {
+			return "", "", errors.Errorf("failed to find secretKey %q for rgw admin ops in secret %q", rgwAdminOpsUserSecretKey, RGWAdminOpsUserSecretName)
+		}
+
+		// Set the keys for further usage
+		return string(accessKey), string(secretKey), nil
+	}
+
+	// Fetch the admin ops user locally
+	userConfig := ObjectUser{
+		UserID:      RGWAdminOpsUserSecretName,
+		DisplayName: &rgwAdminOpsUserDisplayName,
+		SystemUser:  true,
```

Comment:
@leseb : IMO it is better to add the `capabilities` for `AdminOpsUser` than setting it into `SystemUser`. AFAIK `SystemUser` is used to address special cases like `dashboard` or `mutlisite-sync` and not exposed for external S3 use. By setting the `SystemUser` we might end up having certain limitations

Reply:
What about using `admin` then?

Reply:
Sorry, I am not sure about the capabilities of `--admin` flag in the user. The [doc](https://docs.ceph.com/en/latest/radosgw/admin/#add-remove-admin-capabilities) mentions adding admin caps via `caps` options for a user

Reply:
I think the doc is outdated but the `--admin` seems pretty clear to me that it will enable admin caps to a given user. That's probably a way to avoid passing all the caps.

Reply:
Okay if it working as expected then I am not against using it

Reply:
I'm gonna use `caps` so we have a clear understanding of the permissions for that user.